packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,276 @@
+      }
+    | {
+        accessibilityLabel: string,
+        label: string,
+        onClick: $ElementType<ElementConfig<typeof Button>, 'onClick'>,
+        role?: 'button',
+        size?: $ElementType<ElementConfig<typeof Button>, 'size'>,
+      },
+  /**
+   * An optional thumbnail to display next to the text.
+   */
+  thumbnail?:
+    | { image: Element<typeof Image> }
+    | { avatar: Element<typeof Avatar> }
+    | { icon: Element<typeof Icon> },
+  /**
+   *  Heading of BannerOverlay. Content should be [localized](https://gestalt.pinterest.systems/web/banneroverlay#Localization). See the [Text variant](https://gestalt.pinterest.systems/web/banneroverlay#Text) to learn more.
+   */
+  title?: string | Element<typeof Text>,
+  /**
+   *  zIndex of BannerOverlay. See the [zIndex guidelines](https://gestalt.pinterest.systems/web/banneroverlay#zIndex) to learn more.
+   */
+  zIndex?: Indexable,
+};
+
+/**
+ * [BannerOverlays](https://gestalt.pinterest.systems/web/banneroverlay) displays short educational messages when users have performed actions that indicate some intent, such as a related pin tap or idea pin swipe. It is scrim-less, meaning users can scroll content underneath without having a wash layer on top of the content.
+ *
+ * BannerOverlay is a sticky component triggered by scroll, on tap or long-press, placed at the bottom of the screen.
+ * ![BannerOverlay light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/BannerOverlay.spec.mjs-snapshots/BannerOverlay-chromium-darwin.png)
+ */
+export default function BannerOverlay({
+  message,
+  title,
+  dismissButton,
+  helperLink,
+  primaryAction,
+  offsetTop = 130,
+  offsetBottom = 24,
+  thumbnail,
+  zIndex,
+}: Props): ReactNode {
+  const { name: colorSchemeName } = useColorScheme();
+  const isDarkMode = colorSchemeName === 'darkMode';
+
+  const isMobileWidth = useDeviceType() === 'mobile';
+
+  let messageTextElement: Element<'span'> | string;
+
+  if (typeof message === 'string') {
+    messageTextElement = message;
+  }
+  if (typeof title === 'string') {
+    messageTextElement = title;
+  }
+
+  // If `message` is a Text component, we need to override any text colors within to ensure they all match
+  const messageIsTextNode =
+    typeof message !== 'string' &&
+    Children.only<Element<typeof Text>>(message).type.displayName === 'Text';
+
+  if (messageIsTextNode) {
+    const textColorOverrideStyles = isDarkMode
+      ? styles.textColorOverrideDark
+      : styles.textColorOverrideLight;
+
+    messageTextElement = <span className={textColorOverrideStyles}>{message}</span>;
+  }
+
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('BannerOverlay');
+
+  const { lightModeBackground, darkModeBackground, textColor } = DEFAULT_COLORS;
+
+  const dismissButtonComponent = dismissButton && (
+    <Flex.Item flex="none" alignSelf={isMobileWidth ? 'end' : 'center'}>
```

Comment:
consider narrow viewports web... meaning when is desktop BUT we are in narrow window. You might want to play with 

display: none smDisplay: flex etc.